### PR TITLE
[GNA] Fixed concat axis checks

### PR DESCRIPTION
**Pull request by mryzhov**
### Details:
 - Fixes false-positive errors when checking concat axis.

### Tickets:
 - *114093*

### CODE REVIEW

File: src/plugins/intel_gna/src/backend/gna_limitations.cpp [hunk 1]
```diff
@@ -999,9 +1047,13 @@ void Limitations::check_all_ops_supported(const std::shared_ptr<ov::Model>& mode
...
-        if (!is_op_supported(op, gna_precision, true)) {
-            error << "The plugin does not support layer " << op->get_friendly_name() << " (type " << op->get_type_name()
-                  << ")!" << std::endl;
+        try {
+            if (!is_op_supported(op, gna_precision, true)) {
+                error << "The plugin does not support layer " << op->get_friendly_name() << " (type "
+                      << op->get_type_name() << ")!" << std::endl;
+            }
+        } catch (const InferenceEngine::GeneralError& e) {
+            error << e.what() << std::endl;
```
Comment (marcinkusm) [inline src/plugins/intel_gna/src/backend/gna_limitations.cpp:1056]: why do we want to catch this exception? 
do you want to change exception type?

File: src/plugins/intel_gna/tests/functional/shared_tests_instances/subgraph_tests/permute_concat_concat_permute.cpp [hunk 2]
```diff
@@ -24,10 +23,4 @@ INSTANTIATE_TEST_SUITE_P(smoke_permute_concat_concat_permute,
                                             ::testing::Values(CommonTestUtils::DEVICE_GNA)),
                          PermuteConcatConcatPermute::getTestCaseName);
 
-INSTANTIATE_TEST_SUITE_P(smoke_permute_concat_concat_permute,
```
Comment (marcinkusm) [inline src/plugins/intel_gna/tests/functional/shared_tests_instances/subgraph_tests/permute_concat_concat_permute.cpp:27]: why do we remove this tests?

**Review by marcinkusm — APPROVED** [review]
As far as I understood change looks fine.

File: src/plugins/intel_gna/src/backend/gna_limitations.cpp (hunk 1 above)
Comment (mryzhov) [reply src/plugins/intel_gna/src/backend/gna_limitations.cpp:1056]: No, I want to catch all exceptions and list all of them later

File: src/plugins/intel_gna/tests/functional/shared_tests_instances/subgraph_tests/permute_concat_concat_permute.cpp (hunk 2 above)
Comment (mryzhov) [reply src/plugins/intel_gna/tests/functional/shared_tests_instances/subgraph_tests/permute_concat_concat_permute.cpp:27]: I moved all negative cases to positive test  (we support all of them now), and we could not emulate negative case now  without changing the shared test itself

File: src/plugins/intel_gna/src/common/graph_utils.hpp [hunk 3]
```diff
@@ -244,6 +244,50 @@ inline bool is_activation(const std::shared_ptr<ngraph::Node>& node) noexcept {
...
 }
 
+inline bool is_non_functional(const std::shared_ptr<ov::Node>& node) {
+    return std::dynamic_pointer_cast<ov::opset12::Reshape>(node) != nullptr ||
+           std::dynamic_pointer_cast<ov::opset12::Squeeze>(node) != nullptr ||
+           std::dynamic_pointer_cast<ov::opset12::Unsqueeze>(node) != nullptr ||
+           std::dynamic_pointer_cast<ov::opset12::FakeQuantize>(node) != nullptr;
+}
+
+inline bool is_parameter(const std::shared_ptr<ov::Node>& node) {
```
Comment (marcinkacprzak) [inline src/plugins/intel_gna/src/common/graph_utils.hpp:254]: is_parameter() is also defined in op_types.cpp

File: src/plugins/intel_gna/src/common/graph_utils.hpp [hunk 4]
```diff
@@ -244,6 +244,50 @@ inline bool is_activation(const std::shared_ptr<ngraph::Node>& node) noexcept {
...
+
+inline bool is_relu(const std::shared_ptr<ov::Node>& node) {
+    return std::dynamic_pointer_cast<ov::opset12::Relu>(node) != nullptr;
+}
+
+inline bool is_matmul(const std::shared_ptr<ov::Node>& node) {
+    return std::dynamic_pointer_cast<ov::opset12::MatMul>(node) != nullptr;
+}
+
+inline bool is_constant(const std::shared_ptr<ov::Node>& node) {
```
Comment (marcinkacprzak) [inline src/plugins/intel_gna/src/common/graph_utils.hpp:270]: is_constant() is also defined in op_types.cpp.

File: src/plugins/intel_gna/src/common/graph_utils.hpp (hunk 3 above)
Comment (mryzhov) [reply src/plugins/intel_gna/src/common/graph_utils.hpp]: fixed

File: src/plugins/intel_gna/src/common/graph_utils.hpp (hunk 4 above)
Comment (mryzhov) [reply src/plugins/intel_gna/src/common/graph_utils.hpp]: fixed

**Review by marcinkacprzak — APPROVED** [review]

**Pull request merged**